src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -660,15 +660,19 @@ def set_up_create_containerapp_yaml(self, name, file_name):
             del self.containerapp_def["workloadProfileName"]
 
         # Validate managed environment
+        env_id = self.containerapp_def["properties"]['environmentId']
+        env_info = None
+        if self.get_argument_managed_env():
+            if not self.get_argument_disable_warnings() and env_id is not None and env_id != self.get_argument_managed_env():
```

Comment:
it is unnecessary to re-write the warning here?

Reply:
Will consider fixing this for all input args in future.